engine/executor/server.go

```diff
@@ -240,7 +241,7 @@ func (s *Server) PreDispatchTask(ctx context.Context, req *pb.PreDispatchTaskReq
 		// "Use Aborted if the client should retry at a higher-level".
 		// Failure to make task is usually a problem that the business logic
 		// should be notified of.
-		return nil, status.Error(codes.Aborted, err.Error())
+		return nil, registry.GlobalWorkerRegistry().AnalyzeError(err, workerType)
```

Comment:
Can we extract the raw business error from framework wrapped error and pass it to business `IsRetryableError`, rather than use regexp to convert to business error from error messsage?

And seems `makeTask` will not raise errors of DM type, because DM business logic begins to be called after InitImpl

Reply:
> And seems makeTask will not raise errors of DM type, because DM business logic begins to be called after InitImpl

Yes, `makeTask` mainly contains API from registry factory, including `DeserializeConfig` and `NewWorkerImpl`, which covers the error in create worker part.

Other DM business logic such as `Init` and `Tick` are called from the runtime level asynchronously, we need to find a way to communicate between executor and master to reach an agreement about the `fast fail` of a job. That's why we reuse the regulation of DM error or PingCAP rfc error to ship this information.

Reply:
> Can we extract the raw business error from framework wrapped error and pass it to business IsRetryableError, rather than use regexp to convert to business error from error messsage?
Of course, can do it here.

Reply:
addressed, ptal